src/compiler/compile-props.js

```diff
@@ -122,6 +123,9 @@ export function compileProps (el, propOptions, vm) {
     } else if ((value = getAttr(el, attr)) !== null) {
       // has literal binding!
       prop.raw = value
+    } else if (propsData && (value = propsData[attr]) !== null) {
```

Comment:
True, that makes more sense. then we should just replace `attr` with `name` here.